### PR TITLE
Product card

### PR DESCRIPTION
**Pull request by abcdesign1**
- [DEMO LINK](https://vadym4che.github.io/layout_product-cards/)
- [TEST REPORT LINK](https://vadym4che.github.io/layout_product-cards/report/html_report/)

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -1,13 +1,43 @@
...
+    <meta name="viewport"
+          content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
+    <title>Product cards</title>
+    <link rel="stylesheet" href="./style.css">
+  </head>
+  <body>
+    <div class="card" data-qa="card">
+      <img
+        src="./images/imac.jpeg"
+        alt="product picture"
```
Comment (MarkBabich) [inline src/index.html:14]: Its better to choose alt like in p.13 checklist

File: src/style.css [hunk 2]
```diff
@@ -0,0 +1,143 @@
+@import url("https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap");
+
```
Comment (MarkBabich) [inline src/style.css:2]: You should create separate file for each BEM block. You need folder "blocks" with files stars.css and card.css, and separate files for variables and resets.

File: src/style.css [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  box-sizing: border-box;
+  background: #fff;
+  border: 1px solid #f3f3f3;
+  border-radius: 5px;
+  width: 200px;
+  height: 408px;
+  display: flex;
+  flex-direction: column;
+  flex-wrap: wrap;
+  overflow: hidden;
```
Comment (MarkBabich) [inline src/style.css:24]: Do we really need overflow? (sorry if what)

File: src/style.css [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+  font-family: "Roboto", sans-serif;
+}
+
+.card {
+  box-sizing: border-box;
+  background: #fff;
+  border: 1px solid #f3f3f3;
+  border-radius: 5px;
+  width: 200px;
+  height: 408px;
```
Comment (MarkBabich) [inline src/style.css:20]: The height of the block should be set by the content

File: src/style.css [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+  -webkit-border-radius: 5px;
+  -moz-border-radius: 5px;
+  -ms-border-radius: 5px;
+  -o-border-radius: 5px;
+  margin: 17px 16px 0;
+}
+
+.card__buy-button:hover {
+  background-color: #fff;
+  border: 1px solid var(--blue-button);
```
Comment (MarkBabich) [inline src/style.css:141]: Don't add new border to the element on hover. Add default transparent border of the same width, and change just color on :hover

**Review by MarkBabich — CHANGES_REQUESTED** [review]
Great  job. It remains only to correct a little!

File: src/index.html (hunk 1 above)
Comment (vadym4che) [reply src/index.html:14]: fixed

File: src/style.css (hunk 2 above)
Comment (vadym4che) [reply src/style.css:2]: fixed

File: src/style.css (hunk 3 above)
Comment (vadym4che) [reply src/style.css:24]: fixed
thank you for your suggestion

File: src/style.css (hunk 4 above)
Comment (vadym4che) [reply src/style.css:20]: fixed - nothing has been broken

File: src/style.css (hunk 5 above)
Comment (vadym4che) [reply src/style.css:141]: fixed

**Review by MarkBabich — APPROVED** [review]

File: src/index.html [hunk 6]
```diff
@@ -1,13 +1,64 @@
...
+
+        <span class="card__price-value">$2,199</span>
+      </div>
+
+      <a
+        class="card__buy-button"
+        data-qa="hover"
+        href="#">
+        BUY
+      </a>
```
Comment (Andr1yK) [inline src/index.html:61]:
```suggestion
      <a
        class="card__buy-button"
        data-qa="hover"
        href="#"
      >
        BUY
      </a>
```

File: src/css/blocks/card.css [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  width: 160px;
+  height: 134px;
+  margin: 32px 19px 40px;
+}
+
+.card__title {
+  width: 100%;
+  height: 36px;
+  margin: 0 16px;
+  max-width: 166px;
```
Comment (Andr1yK) [inline src/css/blocks/card.css:22]: wow.. why do we need max-width for the title?

Avoid coping styles from Figma🥲

File: src/css/blocks/card.css [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+.card__price-value {
+  font-weight: 700;
+  font-size: 16px;
+  line-height: 16px;
+  color: var(--dark-text);
+}
+
+.card__buy-button {
+  box-sizing: border-box;
+  width: 166px;
```
Comment (Andr1yK) [inline src/css/blocks/card.css:63]: and here

File: src/index.html [hunk 9]
```diff
@@ -1,13 +1,64 @@
...
+    >
+    <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap" rel="stylesheet">
+    <link rel="stylesheet" href="./css/style.css">
+  </head>
+  <body>
+    <div class="card" data-qa="card">
+
+      <img
+        src="./images/imac.jpeg"
+        alt="APPLE A1419 iMac 27&quot; Retina 5K Monoblock (MNED2UA/A)"
```
[the remark this answers is not in the document]
Comment (vadym4che) [reply src/index.html:22]: fixed

File: src/index.html (hunk 6 above)
Comment (vadym4che) [reply src/index.html:61]: fixed

File: src/css/blocks/card.css (hunk 7 above)
Comment (vadym4che) [reply src/css/blocks/card.css:22]: fixed

File: src/index.html [hunk 10]
```diff
@@ -1,13 +1,64 @@
...
+        <div class="stars__star"></div>
+
+        <div class="stars__star"></div>
+
+        <div class="stars__star"></div>
+
+        <span class="stars__reviews">
+          Reviews: 5
+        </span>
+      </div>
```
[the remark this answers is not in the document]
Comment (vadym4che) [reply src/index.html:48]: fixed

File: src/index.html [hunk 11]
```diff
@@ -1,13 +1,64 @@
...
+        <div class="stars__star"></div>
+
+        <div class="stars__star"></div>
+
+        <div class="stars__star"></div>
+
+        <div class="stars__star"></div>
+
+        <div class="stars__star"></div>
+
```
Comment (Andr1yK) [inline src/index.html:43]:
```suggestion
        <div class="stars__star"></div>
        <div class="stars__star"></div>
        <div class="stars__star"></div>
        <div class="stars__star"></div>
        <div class="stars__star"></div>
```

File: src/css/blocks/stars.css [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+.stars--1 .stars__star:first-child,
+.stars--2 .stars__star:first-child,
+.stars--2 .stars__star:nth-child(2),
+.stars--4 .stars__star:not(:last-child),
+.stars__block .stars--5 .stars__star,
+.stars--3 .stars__star:first-child,
+.stars--3 .stars__star:nth-child(2),
+.stars--3 .stars__star:nth-child(3) {
+  background-image: url("./../../images/star-active.svg");
+}
```
Comment (Andr1yK) [inline src/css/blocks/stars.css:32]: let's simplify it

File: src/css/blocks/card.css [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+  display: flex;
+  align-items: center;
+  justify-content: center;
+
+  border: 1px solid transparent;
+  border-radius: 5px;
+  -webkit-border-radius: 5px;
+  -moz-border-radius: 5px;
+  -ms-border-radius: 5px;
+  -o-border-radius: 5px;
```
Comment (Andr1yK) [inline src/css/blocks/card.css:79]: underhood webpack can do it automatically

**Review by Andr1yK — CHANGES_REQUESTED** [review]

File: src/css/blocks/stars.css (hunk 12 above)
Comment (vadym4che) [reply src/css/blocks/stars.css:32]: simplified

File: src/index.html (hunk 11 above)
Comment (vadym4che) [reply src/index.html:43]: spacing removed

File: src/css/blocks/card.css (hunk 8 above)
Comment (vadym4che) [reply src/css/blocks/card.css:59]: fixed

File: src/css/blocks/card.css (hunk 13 above)
Comment (vadym4che) [reply src/css/blocks/card.css:79]: removed

File: src/css/blocks/card.css [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+.card__picture {
+  width: 160px;
+  height: 134px;
+  margin: 32px 19px 40px;
+}
+
+.card__title {
+  height: 36px;
```
Comment (Andr1yK) [inline src/css/blocks/card.css:19]: What happens if the header contains many lines?

🫠

image

File: src/css/blocks/stars.css [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+.stars--4 .stars__star:nth-child(-n+4) {
+  background-image: url("./../../images/star-active.svg");
+}
+
+.reviews {
+  display: inline-block;
+  font-weight: 400;
+  font-size: 10px;
+  margin-left: 22px;
+}
```
Comment (Andr1yK) [inline src/css/blocks/stars.css:36]: We can't set margins for blocks

https://en.bem.info/methodology/css/#external-geometry-and-positioning

File: src/css/blocks/card.css [hunk 16]
```diff
@@ -0,0 +1,83 @@
+.card {
+  box-sizing: border-box;
```
[the remark this answers is not in the document]
Comment (vadym4che) [reply src/css/blocks/card.css:2]: has taken into account

File: src/css/blocks/card.css (hunk 14 above)
Comment (vadym4che) [reply src/css/blocks/card.css]: fixed

File: src/css/blocks/stars.css (hunk 15 above)
Comment (vadym4che) [reply src/css/blocks/stars.css]: cleaned up

**Review by sergii-nosachenko — APPROVED** [review]
Good job 👍

**Pull request closed without merging**